src/SonataCoreBundle.php

```diff
@@ -29,8 +29,15 @@ public function build(ContainerBuilder $container)
         $container->addCompilerPass(new StatusRendererCompilerPass());
         $container->addCompilerPass(new AdapterCompilerPass());
 
+        $formMappingIsEnabled = true;
+        foreach ($container->getExtensionConfig('sonata_core') as $config) {
+            if (isset($config['form']['mapping']['enabled'])) {
```

Comment:
It does not work for me.
How can it work if `build` method is called before extension config is loaded?

Reply:
I don't know, is it really called before, after, or does it depend on something? Couldn't find any docs on this.

Reply:
Do you try to fix it with no ability to debug? :)

Reply:
Yes :'(

Reply:
Plus some people told me it worked :P

Reply:
> Plus some people told me it worked :P

Trust no one.

Reply:
Yeah I trusted 2 people here:

- one who told me that would work
- one who told me it worked

But in my defense it probably works... sometimes?

Reply:
Anyway... does anyone have another idea? Or an idea for a patch that would make this work reliably?

Reply:
I propose to use the composition instead of the inheritance. Therefore deprecated class won't initialize and deprecation message won't appear.
Next `FormFactoryCompilerPass` can get config (config can be got in compiler pass) and if mapping is enabled creates `Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler` and calls `process`.

Reply:
See #490